diff -uNdr firstboot-1.117-original/modules/create_user.py firstboot-1.117/modules/create_user.py
--- firstboot-1.117-original/modules/create_user.py	2011-02-18 12:39:54.000000000 +0300
+++ firstboot-1.117/modules/create_user.py	2011-03-23 02:03:52.972012417 +0300
@@ -232,11 +232,26 @@
 
         self.admin.setpassUser(userEnt, self.passwordEntry.get_text(), 0)
 
-        # add user to wheel group
-        if self.is_admin.get_active():
+        # add user to wheel group # hacked by Russian Fedora ;)
+        '''if self.is_admin.get_active():
             wheelEnt = self.admin.lookupGroupByName("wheel")
             wheelEnt.add(libuser.MEMBERNAME, username)
-            self.admin.modifyGroup(wheelEnt)
+            self.admin.modifyGroup(wheelEnt)'''
+        # FIXME: elemc: Her is radiobutton active states
+
+        if self.ch1.get_active() == 0:
+            pass
+        else:
+            sudo_hook_name = '/etc/sudoers.d/%s' % username
+            sudo_hood = open(sudo_hook_name, 'w')
+            if self.ch1.get_active() == 1:
+                sudo_hood.write('%s ALL=(ALL) ALL\n' % username)
+            elif self.ch1.get_active() == 2:
+                sudo_hood.write('%s ALL=(ALL) NOPASSWD: ALL\n' % username)
+
+            sudo_hood.close()
+            os.chmod(sudo_hook_name, 0440)
+        
 
         return RESULT_SUCCESS
 
@@ -278,7 +293,7 @@
 
         self.vbox.pack_start(label, False, True)
 
-        table = gtk.Table(3, 4)
+        table = gtk.Table(3, 5)
         table.set_row_spacings(6)
         table.set_col_spacings(6)
 
@@ -313,9 +328,12 @@
         table.attach(self.strengthLabel, 2, 3, 2, 3, gtk.FILL)
         table.attach(self.confirmIcon, 2, 3, 3, 4, gtk.FILL)
 
-        self.is_admin = gtk.CheckButton(_("Add to Administrators group"))
-        self.is_admin.set_alignment(0.0, 0.5)
-        table.attach(self.is_admin, 2, 3, 1, 2, gtk.FILL)
+        self.ch1 = gtk.combo_box_new_text()
+        self.ch1.append_text( _('Don\'t use sudo') )
+        self.ch1.append_text( _('Use sudo with password prompt') )
+        self.ch1.append_text( _('Use sudo without password prompt') )
+        self.ch1.set_active(0)
+        table.attach(self.ch1, 0, 2, 4, 5, gtk.FILL, gtk.FILL, 5)
 
         self.vbox.pack_start(table, False)
 
diff -uNdr firstboot-1.117-original/po/ru.po firstboot-1.117/po/ru.po
--- firstboot-1.117-original/po/ru.po	2011-02-18 12:39:54.000000000 +0300
+++ firstboot-1.117/po/ru.po	2011-03-23 02:02:56.364703202 +0300
@@ -237,6 +237,18 @@
 msgid "Confir_m Password:"
 msgstr "По_дтвердите пароль:"
 
+#: ../modules/create_user.py:332
+msgid "Don't use sudo"
+msgstr "Не использовать sudo"
+
+#: ../modules/create_user.py:333
+msgid "Use sudo with password prompt"
+msgstr "Использовать sudo с запросом пароля"
+
+#: ../modules/create_user.py:334
+msgid "Use sudo without password prompt"
+msgstr "Использовать sudo без запроса пароля"
+
 #: ../modules/create_user.py:273
 msgid ""
 "If you need to use network authentication, such as Kerberos or NIS, please "
@@ -259,16 +271,16 @@
 msgid "_Advanced..."
 msgstr "_Дополнительно..."
 
-#: ../modules/create_user.py:338
+#: ../modules/create_user.py:405
 msgid "Please wait"
 msgstr "Подождите"
 
-#: ../modules/create_user.py:375
+#: ../modules/create_user.py:442
 #, python-format
 msgid "Weak password: %s"
 msgstr "Слабый пароль: %s"
 
-#: ../modules/create_user.py:379
+#: ../modules/create_user.py:446
 msgid "Password OK"
 msgstr "Пароль ОК"
 
