diff -up firstboot-1.113/firstboot/xfrontend.py.kwin firstboot-1.113/firstboot/xfrontend.py
--- firstboot-1.113/firstboot/xfrontend.py.kwin	2010-08-30 05:57:44.000000000 -0500
+++ firstboot-1.113/firstboot/xfrontend.py	2010-09-15 14:26:03.900021796 -0500
@@ -42,7 +42,9 @@ class XFrontEnd:
         self._wm_pid = os.fork()
 
         if not self._wm_pid:
-            path = "/usr/bin/metacity"
+            paths = ["/usr/bin/metacity", "/usr/bin/kwin", "/usr/bin/xfwm4", "/usr/bin/openbox"]
+            for path in paths:
+                if os.access(path, os.X_OK): break 
             args = [path, "--display", os.environ["DISPLAY"]]
             os.execvp(path, args)
 
