diff -uNdr firstboot-1.113.orig/modules/create_user.py firstboot-1.113/modules/create_user.py
--- firstboot-1.113.orig/modules/create_user.py	2011-02-11 17:56:37.000000000 +0300
+++ firstboot-1.113/modules/create_user.py	2011-02-11 20:50:38.689190985 +0300
@@ -208,9 +208,22 @@
 
         self.admin.setpassUser(userEnt, self.passwordEntry.get_text(), 0)
 
-        if self.ch1.get_property('sensitive') and self.ch1.state == 1:
-            os.system('echo "' + username + ' ALL=(ALL) ALL" >> /etc/sudoers.d/' + username)
-            os.system('chmod 0440 /etc/sudoers.d/' + username)
+        # FIXME: elemc: Her is radiobutton active states
+        #if self.ch1.get_property('sensitive') and self.ch1.state == 1:
+        #    os.system('echo "' + username + ' ALL=(ALL) ALL" >> /etc/sudoers.d/' + username)
+        #    os.system('chmod 0440 /etc/sudoers.d/' + username)
+        if self.ch1.get_active() == 0:
+            pass
+        else:
+            sudo_hook_name = '/etc/sudoers.d/%s' % username
+            sudo_hood = open(sudo_hook_name, 'w')
+            if self.ch1.get_active() == 1:
+                sudo_hood.write('%s ALL=(ALL) ALL\n' % username)
+            elif self.ch1.get_active() == 2:
+                sudo_hood.write('%s ALL=(ALL) NOPASSWD: ALL\n' % username)
+
+            sudo_hood.close()
+            os.chmod(sudo_hook_name, 0440)
 
         if self.ch2.get_property('sensitive') and self.ch2.state == 1:
             if os.path.exists('/etc/gdm/custom.conf'):
@@ -255,7 +268,7 @@
 
         self.vbox.pack_start(label, False, True)
 
-        table = gtk.Table(2, 4)
+        table = gtk.Table(2, 5)
         table.set_row_spacings(6)
         table.set_col_spacings(6)
         label = gtk.Label(_("_Username:"))
@@ -288,10 +301,18 @@
 
         self.vbox.pack_start(table, False)
 
-        self.vbox.pack_start(label, False, False, 10)
+        #self.vbox.pack_start(label, False, False, 10)
 
-        self.ch1 = gtk.CheckButton(label=_('Allow this user to use sudo'))
-        self.vbox.pack_start(self.ch1, False, True, 1)
+        # FIXME: elemc: change to 3 radio button with choice: Without sudo (defual), use sudo and use sudo without password prompt
+        #self.ch1 = gtk.CheckButton(label=_('Allow this user to use sudo'))
+        self.ch1 = gtk.combo_box_new_text()
+        self.ch1.append_text( _('Don\'t use sudo') )
+        self.ch1.append_text( _('Use sudo with password prompt') )
+        self.ch1.append_text( _('Use sudo without password prompt') )
+        self.ch1.set_active(0)
+        table.attach(self.ch1, 0, 2, 4, 5, gtk.FILL, gtk.FILL, 5)
+        #self.vbox.pack_start(self.ch1, False, False, 1)
+        #self.vbox.pack_start(self.ch1w_sudo_np, False, True, 1)
 
         self.ch2 = gtk.CheckButton(label=_('Enable Autologin'))
         self.vbox.pack_start(self.ch2, False, True, 1)
diff -uNdr firstboot-1.113.orig/po/ru.po firstboot-1.113/po/ru.po
--- firstboot-1.113.orig/po/ru.po	2011-02-11 17:56:37.000000000 +0300
+++ firstboot-1.113/po/ru.po	2011-02-11 20:38:04.157757528 +0300
@@ -245,31 +245,39 @@
 msgid "Confir_m Password:"
 msgstr "По_дтвердите пароль:"
 
-#: ../modules/create_user.py:293
-msgid "Allow this user to use sudo"
-msgstr "Разрешить этому пользователю использовать sudo"
+#: ../modules/create_user.py:309
+msgid "Don't use sudo"
+msgstr "Не использовать sudo"
 
-#: ../modules/create_user.py:296
+#: ../modules/create_user.py:310
+msgid "Use sudo with password prompt"
+msgstr "Использовать sudo с запросом пароля"
+
+#: ../modules/create_user.py:311
+msgid "Use sudo without password prompt"
+msgstr "Использовать sudo без запроса пароля"
+
+#: ../modules/create_user.py:317
 msgid "Enable Autologin"
 msgstr "Разрешить автоматический вход в систему"
 
-#: ../modules/create_user.py:302
+#: ../modules/create_user.py:323
 msgid "If you need to use network authentication, such as Kerberos or NIS, please click the Use Network Login button."
 msgstr "Если требуется использовать проверку подлинности по сети, например Kerberos или NIS, нажмите кнопку «Сетевая авторизация»."
 
-#: ../modules/create_user.py:311
+#: ../modules/create_user.py:332
 msgid "Use Network _Login..."
 msgstr "Сетевая аут_ентификация..."
 
-#: ../modules/create_user.py:319
+#: ../modules/create_user.py:340
 msgid "If you need more control when creating the user (specifying home directory, and/or UID), please click the Advanced button."
 msgstr "Если вам необходимо больше контроля, когда создаёте пользователя (указать домашний каталог и/или UID), пожалуйста, щёлкните по кнопке Дополнительно."
 
-#: ../modules/create_user.py:329
+#: ../modules/create_user.py:350
 msgid "_Advanced..."
 msgstr "_Дополнительно..."
 
-#: ../modules/create_user.py:308
+#: ../modules/create_user.py:388
 msgid "Please wait"
 msgstr "Подождите"
 
