diff -urN firstboot-1.110.11.orig/modules/create_user.py firstboot-1.110.11/modules/create_user.py
--- firstboot-1.110.11.orig/modules/create_user.py	2012-01-23 15:24:43.299444985 +0400
+++ firstboot-1.110.11/modules/create_user.py	2012-01-23 15:26:22.231190773 +0400
@@ -229,6 +229,18 @@
 
         self.admin.setpassUser(userEnt, self.passwordEntry.get_text(), 0)
 
+        if self.ch1.get_active() == 0:                                                                                                                                                                                            
+            pass                                                                                                                                                                                                                  
+        else:                                                                                                                                                                                                                     
+            sudo_hook_name = '/etc/sudoers.d/%s' % username
+            sudo_hood = open(sudo_hook_name, 'w')
+            if self.ch1.get_active() == 1:
+                sudo_hood.write('%s ALL=(ALL) ALL\n' % username)
+            elif self.ch1.get_active() == 2:
+                sudo_hood.write('%s ALL=(ALL) NOPASSWD: ALL\n' % username)
+            sudo_hood.close()
+            os.chmod(sudo_hook_name, 0440)
+
         return RESULT_SUCCESS
 
     def createScreen(self):
@@ -282,6 +294,13 @@
         table.attach(label, 0, 1, 3, 4, gtk.FILL)
         table.attach(self.confirmEntry, 1, 2, 3, 4, gtk.SHRINK, gtk.FILL, 5)
 
+        self.ch1 = gtk.combo_box_new_text()
+        self.ch1.append_text( _('Don\'t use sudo') )
+        self.ch1.append_text( _('Use sudo with password prompt') )
+        self.ch1.append_text( _('Use sudo without password prompt') )
+        self.ch1.set_active(0)
+        table.attach(self.ch1, 0, 2, 4, 5, gtk.FILL, gtk.FILL, 5)
+
         self.vbox.pack_start(table, False)
 
         label = gtk.Label(_("If you need to use network authentication, such as Kerberos or NIS, "
diff -urN firstboot-1.110.11.orig/po/ru.po firstboot-1.110.11/po/ru.po
--- firstboot-1.110.11.orig/po/ru.po	2012-01-23 15:24:43.297444946 +0400
+++ firstboot-1.110.11/po/ru.po	2012-01-23 15:26:55.255832911 +0400
@@ -224,6 +224,18 @@
 msgid "Confir_m Password:"
 msgstr "По_дтвердите пароль:"
 
+#: ../modules/create_user.py:298
+msgid "Don't use sudo"
+msgstr "Не использовать sudo"
+
+#: ../modules/create_user.py:299
+msgid "Use sudo with password prompt"
+msgstr "Использовать sudo с запросом пароля"
+
+#: ../modules/create_user.py:300
+msgid "Use sudo without password prompt"
+msgstr "Использовать sudo без запроса пароля"
+
 #: ../modules/create_user.py:287
 msgid ""
 "If you need to use network authentication, such as Kerberos or NIS, please "
